caluma/caluma_form/models.py

```diff
@@ -208,8 +208,44 @@ class Document(UUIDModel):
     form = models.ForeignKey(
         "caluma_form.Form", on_delete=models.DO_NOTHING, related_name="documents"
     )
+    source = models.ForeignKey(
+        "self",
+        blank=True,
+        null=True,
+        help_text="Reference this document has been copied from",
+        related_name="+",
+        on_delete=models.SET_NULL,
+    )
     meta = JSONField(default=dict)
 
+    def copy(self, family=None):
+        """Create a copy including all its answers."""
+
+        new_document = self.__class__.objects.create(
+            form=self.form, meta=dict(self.meta), source=self, family=family
+        )
+        if not family:
+            family = new_document.family
+
+        # copy all answers
+        for source_answer in self.answers.all():
+            new_answer = new_document.answers.create(
+                question=source_answer.question,
+                value=source_answer.value,
+                meta=dict(source_answer.meta),
+                date=source_answer.date,
+            )
+
+            # TableAnswer: copy AnswerDocument too
+            for answer_doc in AnswerDocument.objects.filter(answer=source_answer):
+                new_doc = answer_doc.document.copy(family=family)
+
+                AnswerDocument.objects.create(
+                    answer=new_answer, document=new_doc, sort=answer_doc.sort
+                )
+
+        return new_document
```

Comment:
What about file answers, their containing file objects and the files in Minio?

Reply:
They are intentionally ignored for the time being...

Reply:
So we should probably not copy them at all to prevent accidentally changing/deleting file answers from the original?

Reply:
Excluded FileAnswers for now

Reply:
Isn't this some really unexpected/surprising behaviour? The whole document is copied, except for files?

Reply:
Well I added this as a TODO, but your point stands of course

Reply:
Can we add this to the graphiQL docs? Something like "Warning: Copying of file answers is currently not supported"

Reply:
Sorry, I don't really understand this discussion. Why don't we just also copy the files? Doesn't seem that difficult...

Reply:
This decision was made purely because we don't need it for our current use case. If it's really that simple, it would be best to implement it of course!

Reply:
IMO it's ok to not implement something we don't need. Filters come to mind.

But I don't think we should partially implement features, just because we don't need all of it right now.

Reply:
I guess copying the caluma FileAnswer is indeed not very hard. What we've been wondering earlier was if we also need to copy the file in minio - which sounds like more effort.

Reply:
Copying the FileAnswer is trivial. However it's very dangerous if we just copy the Answer but not the File. So we end up with two Answers referencing the same file...

Reply:
Yes, the file in minio has to be duplicated too. Removal of a fileanswer also removes the corresponding `File` object. Removing a `File` object also removes the file in minio (in fact it moves the file in minio, so it stays available for historical records.

So if multiple `File` objects reference the same minio file and one of those `File` objects gets deleted, the other `File` objects reference a file that doesn't exist anymore.

Reply:
Right - do you think copying the file inside minio is difficult?

Reply:
Alright we're doing the file copy!

Reply:
Done

Reply:
Looks simple enough 😅